### ISSUE TITLE
syncing orphan blocks before fully synced.

### ISSUE THREAD

**Issue by j4ys0n** [decred/dcrd#1312]
I found an issue on this, and it said that it will add "orphan blocks" and reconcile them later, but my blockchain isn't fully synced and it's not resuming the sync but instead adding orphan blocks...

```2018-06-21 18:39:36.425 [INF] BMGR: Processed 178 blocks in the last 10.03s (1474 transactions, 1757 stake transactions, height 205180, 2018-01-19 08:04:11 +0000 UTC)
2018-06-21 18:39:37.076 [INF] BMGR: Rejected block 00000000000000411542951d35c3b3ec53743356f733b5511820e2296bea2b51 from 158.69.126.156:9108 (outbound): failed to validate input 177f1202f2a4b1dcd76076a57706b29eb2a8cb0d5ba99143a59bcbdf0d3cdf9e:0 which references output 2b56abd7f8c782fcd7768b266109446db117bf352ae765f6bf40c090c8c9ea0f:0 - verify failed (input script bytes 473044022064ed45155231e396e7a702ce71945bb65343cc7191f85774d1df4f5cba91cf5b022043fedfec887b26269210872fa442549747a9c723f8e7d0063e0709a6121feea1012102145f9733076af3b84a05452175ded7245f45fb227a180ecfa2f36acd5c91233220812ef04a8d8819a002138c9d33be68cc71c648acb5e478b9328402eb591a8109514c5d63c02089dcaf6352673bee65b119b33ddf7ba5cbd76408bbfdda8eab2f160dbaab2beb8876a9149be7df505ea9c8c62f3d5e718c3ba2a813ecc31367049805635ab17576a914a93b0d16f425239b9fc80ad8fde23fd24f5507ec6888ac, prev output script bytes a914f5beeea43613cbe64c9716b46e52f5cff21ee3f587)
2018-06-21 18:39:37.080 [INF] CHAN: Adding orphan block 000000000000000230c6012bfc7fafc574c426d4c500d63c1a59b41d5fa176f4 with parent 00000000000000411542951d35c3b3ec53743356f733b5511820e2296bea2b51
```

what should i do?

**Comment by davecgh** [comment]
Looks like you are running an old version of dcrd.  There was [a vote](https://docs.decred.org/getting-started/user-guides/agenda-voting/) to add new non-backwards-compatible features of which the first block that made use of them was [205189](https://explorer.dcrdata.org/block/00000000000000411542951d35c3b3ec53743356f733b5511820e2296bea2b51), which is precisely the block your node is rejecting.

In other words, upgrade.

**Comment by j4ys0n** [comment]
i'm running the docker image.. was that not updated?

**Comment by davecgh** [comment]
There are no officially supported Docker images.

**Comment by j4ys0n** [comment]
this is not supported? https://github.com/decred/dcrdocker

**Comment by davecgh** [comment]
I'll let @dajohi chime in since he maintains those, but afaik those are only for the purposes of having the recipes and testing.  That said, I don't use Docker, so I could be mistaken.

What I can say for sure is that as the current maintainer of the dcrd repo, I absolutely and unequivocally do not recommend nor support the use of Docker images from Dockerhub.  It is just way too easy to inject malicious code and is incredibly dangerous to use.  The officially supported and *signed* binaries are at https://github.com/decred/decred-binaries/releases.

**Comment by j4ys0n** [comment]
awesome. that's all i really need :) thank you!

**Comment by davecgh** [comment]
No problem.  By the way, if you go to https://decred.org/downloads, there is a link to an installer for the CLI tools which sets up `dcrctl` and all too, as well as checks the signatures, if you want to go that way.  If you prefer to just set it up manually using the binaries, that works too.

**Comment by j4ys0n** [comment]
@davecgh almost forgot - do those releases have the insight methods?

**Comment by dajohi** [comment]
@j4ys0n decred/dcrd-mainnet and decred/dcrd-testnet updated to 1.2.0.

**Comment by davecgh** [comment]
No, the Insight API is a separate thing that is currently offered by https://mainnet.decred.org.  There is also work to support the Insight API via [dcrdata](https://github.com/decred/dcrdata), but that work is not complete yet.

As an aside, dcrd offers a [JSON-RPC API](https://github.com/decred/dcrd/blob/master/docs/json_rpc_api.md) which is what insight makes use of to build its own database of information.

I would guess you're asking because you want to setup your own local instances of insight.  The process is currently pretty involved as it involves all of the node.js setup to make the bitcore port work.  In that case, using the (now updated as @dajohi just commented) docker files to build the images yourself is probably the easiest route.  I just don't at all recommend downloading the binaries from Dockerhub for the aforementioned reason, but building them yourself is safe.

**Comment by j4ys0n** [comment]
yeah i usually compile myself but this time i used the docker images to get an example running, because that's usually quick and easy (i'm running like 15 insight instances on different blockchains).

thanks @dajohi! those are insight-ready?

**Comment by j4ys0n** [comment]
looks like it is.. `[INF] INDX: Exists address index is enabled`

**Comment by dajohi** [comment]
also
dcrd-mainnet-insight
dcrd-testnet-insight

with --txindex=1

**Comment by dajohi** [comment]
and a hardcoded user/pass.

but you could use dcrd-mainnet and use the docker run -v option to create your own ~/.dcrd/dcrd.conf

**Comment by j4ys0n** [comment]
thanks guys! appreciate the quick responses. 🙌🏼
